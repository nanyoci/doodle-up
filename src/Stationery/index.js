import React, { useRef, useEffect, useState } from 'react';

import './index.css';

import LightBluePencil from './../assets/pencillightblue.svg'
import RedPencil from './../assets/pencilred.svg'
import OrangePencil from './../assets/pencilorange.svg'
import GreenPencil from './../assets/pencilgreen.svg'
import BluePencil from './../assets/pencilblue.svg'
import PurplePencil from './../assets/pencilpurple.svg'
import PinkPencil from './../assets/pencilpink.svg'
import YellowPencil from './../assets/pencilyellow.svg'
import BlackPencil from './../assets/pencilblack.svg'
import BrownPencil from './../assets/pencilbrown.svg'
import GrayPencil from './../assets/pencilgray.svg'
import DarkGreenPencil from './../assets/pencildarkgreen.svg'
import Eraser from './../assets/eraser.png'

function Stationery(props) {

    const selectColor = (colorSelect) => {
        props.selectColor(colorSelect)
    }


    return (
        <div id="tools">
            <img className="tool" src={LightBluePencil} alt="Black Pencil" onClick={() => selectColor(props.color.LIGHTBLUE)} />
            <img className="tool" src={RedPencil} alt="Red Pencil" onClick={() => selectColor(props.color.RED)} />
            <img className="tool" src={OrangePencil} alt="Orange Pencil" onClick={() => selectColor(props.color.ORANGE)} />
            <img className="tool" src={DarkGreenPencil} alt="Dark Green Pencil" onClick={() => selectColor(props.color.DARKGREEN)} />
            <img className="tool" src={GreenPencil} alt="Green Pencil" onClick={() => selectColor(props.color.GREEN)} />
            <img className="tool" src={BluePencil} alt="Blue Pencil" onClick={() => selectColor(props.color.BLUE)} />
            <img className="tool" src={PurplePencil} alt="Purple Pencil" onClick={() => selectColor(props.color.PURPLE)} />
            <img className="tool" src={PinkPencil} alt="Pink Pencil" onClick={() => selectColor(props.color.PINK)} />
            <img className="tool" src={YellowPencil} alt="Yellow Pencil" onClick={() => selectColor(props.color.YELLOW)} />
            <img className="tool" src={GrayPencil} alt="Gray Pencil" onClick={() => selectColor(props.color.GRAY)} />
            <img className="tool" src={BrownPencil} alt="Brown Pencil" onClick={() => selectColor(props.color.BROWN)} />
            <img className="tool" src={BlackPencil} alt="Black Pencil" onClick={() => selectColor(props.color.BLACK)} />
            <img className="tool" src={Eraser} alt="Eraser" onClick={() => selectColor(props.color.ERASER)} />
        </div>

    );
}

export default Stationery;
