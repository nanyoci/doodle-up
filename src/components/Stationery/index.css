
#tools{
    display: flex;
    justify-content: center;
    /* height: 100%; */
}

.tool{
    /* height: 100%; */
    width: 5vw;
    cursor: pointer;
    
}

