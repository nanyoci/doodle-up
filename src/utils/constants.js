export const API_URL = 'https://doodle-up-backend.herokuapp.com';