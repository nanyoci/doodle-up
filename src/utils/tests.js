import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )

    const renderedComponent = <Provider store={store}>{component}</Provider>;

    return {
        ...render(renderedComponent),
        store
    };
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    const renderedComponent = <Router history={history}>{component}</Router>;

    return {
        ...render(renderedComponent),
        history
    };
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    const history = createMemoryHistory({ initialEntries: [route] });

    const renderedComponent = (
        <Provider store={store}>
            <Router history={history}>{component}</Router>
        </Provider>
    );

    return {
        ...render(renderedComponent),
        history,
        store
    };
}