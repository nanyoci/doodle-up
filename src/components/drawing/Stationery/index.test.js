import React from 'react'
import { cleanup, act } from '@testing-library/react'

import Stationery from './index.js';
import { renderWithReduxRouter } from './../../../utils/tests';

jest.mock('axios');
console.warn = jest.fn()

beforeEach(() => {
    window.HTMLMediaElement.prototype.load = () => { /* do nothing */ };
    window.HTMLMediaElement.prototype.play = () => { /* do nothing */ };
    window.HTMLMediaElement.prototype.pause = () => { /* do nothing */ };
})

afterEach(() => {
    cleanup();
    jest.clearAllMocks();
});


it('should take a snapshot', async () => {
    let asFragment = {};
    await act(async () => { ({ asFragment } = renderWithReduxRouter(<Stationery />)) });
    expect(asFragment()).toMatchSnapshot();
})