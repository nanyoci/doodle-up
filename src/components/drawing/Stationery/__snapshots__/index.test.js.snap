// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    id="tools"
  >
    <img
      alt="Black Pencil"
      class="tool"
      src="pencillightblue.svg"
    />
    <img
      alt="Red Pencil"
      class="tool"
      src="pencilred.svg"
    />
    <img
      alt="Orange Pencil"
      class="tool"
      src="pencilorange.svg"
    />
    <img
      alt="Dark Green Pencil"
      class="tool"
      src="pencildarkgreen.svg"
    />
    <img
      alt="Green Pencil"
      class="tool"
      src="pencilgreen.svg"
    />
    <img
      alt="Blue Pencil"
      class="tool"
      src="pencilblue.svg"
    />
    <img
      alt="Purple Pencil"
      class="tool"
      src="pencilpurple.svg"
    />
    <img
      alt="Pink Pencil"
      class="tool"
      src="pencilpink.svg"
    />
    <img
      alt="Yellow Pencil"
      class="tool"
      src="pencilyellow.svg"
    />
    <img
      alt="Gray Pencil"
      class="tool"
      src="pencilgray.svg"
    />
    <img
      alt="Brown Pencil"
      class="tool"
      src="pencilbrown.svg"
    />
    <img
      alt="Black Pencil"
      class="tool"
      src="pencilblack.svg"
    />
    <img
      alt="Eraser"
      class="tool"
      src="eraser.svg"
    />
  </div>
</DocumentFragment>
`;
